### ISSUE TITLE
[BUG]: Many labels in the popup can obscure the whole screen

### ISSUE THREAD

**Issue by Mitecon** [martinrotter/rssguard#971]
### Brief description of the issue

If you have a lot of labels and either right-click and hover over the 'Labels' entry or click on the little button below all the coloured dots to the left of the article pane, then a list of all labels pops up it's impossible to see all of the labels. I literally have too many labels to display on the screen at once now and there's no way to click 'Next' to see more. I have to escape out of this menu in order to do anything else.

![lots-of-labels-popup](https://github.com/martinrotter/rssguard/assets/125831943/09509cbf-9ffa-4452-abdd-d2118d725acf)

This is especially noticeable if the names of the labels are quite long:

![lots-of-labels-but-even-longer-popup](https://github.com/martinrotter/rssguard/assets/125831943/e254249b-c4c8-4f58-b286-8c4b1b23379f)

### How to reproduce the bug?

As an example:

If I want to delete an article, I can right-click on the article and move my mouse down to the bottom of the menu to click 'Delete articles'. However, if when moving my mouse down and I have a lot of labels, then I'll briefly mouseover the 'Labels' menu and then I'll get a popup menu with all my labels taking up the whole screen as in the screenshots above. Then I can't reach the bottom of the menu anymore as it's completely obscured. I will have to press escape and right-click again but move my mouse to the side and carefully go around the 'Labels' entry in order to click on the option I want. It doesn't help that the 'Labels' submenu is right there at the top, making it one of the first things the mouse cursor will hit.

Also think about people who use larger fonts for accessibility. I created 180 labels to demonstrate all of this (dedication) but you wouldn't even need that many to fill the screen if you're using a larger font.

I propose a menu that is instead limited to one column only and is scrollable. Think of lists in phone apps that are obviously way too long to fit on a small phone screen. The developers have to put arrows at the top and bottom to indicate the list has items above and below and can be scrolled up and down.

Also, to the left-hand side of the article pane is a list of coloured dots for each label. Barely any of my labels fit on there, making it of no use to me. I know I can click on the little button at the bottom to pop up the rest of the labels but then we run into the problem above where they can't all fit on screen and can't be scrolled if the label you wanted to select was the very last one in the list.

Personally, I can't really imagine a time when I would even use this function so I'd like the ability to disable label submenus completely. Especially if it's related somehow to performance issues.

### What was the expected result?

To not have the whole screen obscured.

### What actually happened?

The whole screen was obscured.

### Debug log

This doesn't need a log.

### Operating system and version

* OS: Manjaro
* RSS Guard version: This is not version-specific

**Comment by martinrotter** [comment]
HOnestly, how to deal with this? Maybe put in case of many items those label items into submenus, divided by leading letters?

**Comment by martinrotter** [comment]
FOR  ME:

- make QMenu subclass and make that menu scrollable

ALSO:

https://stackoverflow.com/questions/7121718/create-a-scrollbar-in-a-submenu-qt

**Comment by martinrotter** [comment]
Yes, I will have to implement such a menu almost from scratch, will test tomorrow.

**Comment by l5oukz89** [comment]
i hope a solution will be implemented for that problem

**Comment by larsen0815** [comment]
No idea if this is possible, but maybe add a filter as a text box. The user would start to type some letters and the program would only display the proper labels. Would make it easier selecting from a very long list, I guess.

**Comment by martinrotter** [comment]
<img width="1350" height="848" alt="Image" src="https://github.com/user-attachments/assets/f29d3c46-5a0f-4660-8454-569e70283c26" />

it's happening guys, scrollable menu

**Comment by martinrotter** [comment]
the vertical list will still be there + new "tag" button to show all labels as pageable scrollable menu
